src/canvas.ts

```diff
@@ -0,0 +1,138 @@
+import { Renderer, basePoint } from './utilities/renderer';
```

Comment:
[
  {"start": 67, "end": 107, "message": "Change private context(): CanvasRenderingContext2D {\n    // eslint-disable-next-line @typescript-eslint/no-non-null-assertion\n    const context: CanvasRenderingContext2D = this.element.getContext('2d')!;\n    context.save();\n    context.beginPath();\n    return context;\n  }\n   to\n  private getContext(): CanvasRenderingContext2D {\n      const context: CanvasRenderingContext2D = this.element.getContext('2d');\n      return context!;\n  }\n"},
  {"start": 118, "end": 145, "message": "Change private fill(context: CanvasRenderingContext2D): void {\n    context.fill();\n    context.closePath();\n    context.restore();\n  }\n  to\n  private fill(context: CanvasRenderingContext2D): void {\n      context.fill();\n      context.restore();\n  }\n"},
  {"start": 52, "end": 61, "message": "Change getDrawBuilder(context: CanvasRenderingContext2D) {\n    return {\n      color: (color: string) => this.color(context, color),\n      width: (lineWidth: number) => this.lineWidth(context, lineWidth),\n      stroke: () => this.stroke(context),\n      fill: () => this.fill(context),\n      draw: () => this.fill(context),\n    };\n  }\n to\n  private getDrawBuilder(context: CanvasRenderingContext2D) {\n      return {\n        color: (color: string) => this.color(context, color),\n        width: (lineWidth: number) => this.lineWidth(context, lineWidth),\n        stroke: () => this.stroke(context),\n        fill: () => this.fill(context),\n      };\n  }\n"},
  {"start": 109, "end": 116, "message": "Change circle({ center, radius }: CircleProperties) {\n    const { context }: Canvas = this;\n    context.arc(center.x, center.y, radius, 0, 2 * Math.PI);\n    return this.getDrawBuilder(context);\n  }\n to\n  circle({ center, radius }: CircleProperties) {\n      const context: CanvasRenderingContext2D = this.getContext();\n      context.beginPath();\n      context.arc(center.x, center.y, radius, 0, 2 * Math.PI);\n      return this.getDrawBuilder(context);\n  }\n"},
  {"start": 149, "end": 187, "message": "Change text({ start, text, size }: TextProperties) {\n    const { context }: Canvas = this;\n    context.font = `Consolas ${size}px`;\n    context.fillText(text, start.x, start.y);\n    context.closePath();\n    context.restore();\n  }\n   to\n  text({ start, text, size }: TextProperties) {\n      const context: CanvasRenderingContext2D = this.getContext();\n      context.beginPath();\n      context.textAlign = 'start';\n      context.font = `Consolas ${size}px`;\n      context.fillText(text, start.x, start.y);\n      context.closePath();\n      context.restore();\n  }\n"},
  {"start": 191, "end": 229, "message": "Change image({ image, start, imageStart, size, ratio = 1 }: ImageProperties) {\n    const { context }: Canvas = this;\n    context.drawImage(image, imageStart.x, imageStart.y, size.width, size.height, start.x, start.y, size.width * ratio, size.height * ratio);\n    context.closePath();\n    context.restore();\n  }\n  to\n  image({ image, start, imageStart, size, ratio = 1 }: ImageProperties) {\n      const context: CanvasRenderingContext2D = this.getContext();\n      context.beginPath();\n      context.drawImage(image, imageStart.x, imageStart.y, size.width, size.height, start.x, start.y, size.width * ratio, size.height * ratio);\n      context.closePath();\n      context.restore();\n  }\n"}
]